server/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalTopHits.java

```diff
@@ -184,9 +185,42 @@ protected boolean mustReduceOnSingleInternalAgg() {
     public Object getProperty(List<String> path) {
         if (path.isEmpty()) {
             return this;
+        }
+        if (path.size() != 1) {
+            throw new IllegalArgumentException(
+                "property paths for top_hits ["
+                    + getName()
+                    + "] can only contain a single field in _source, score or sort values, got "
+                    + path
+            );
+        }
+
+        // Supported property prefixes.
+        final String SOURCE = "_source";
+        final String SORT_VALUE = "_sort";
+        final String SCORE = "_score";
+
+        String[] tokens = path.get(0).toLowerCase(Locale.ROOT).split(":|>|\\.");
+        assert searchHits.getHits().length == 1 : "property paths should only resolve against top_hits with size == 1.";
+        SearchHit topHit = searchHits.getAt(0);
+        if (tokens[0].equals(SORT_VALUE)) {
+            Object[] sortValues = topHit.getSortValues();
+            if (sortValues != null) {
+                return sortValues[0];
+            }
+        } else if (tokens[0].equals(SCORE)) {
+            return topHit.getScore();
         } else {
-            throw new IllegalArgumentException("path not supported for [" + getName() + "]: " + path);
+            Map<String, Object> sourceAsMap = topHit.getSourceAsMap();
+            if (sourceAsMap != null) {
+                String field = tokens[0];
+                if (tokens.length == 2 && tokens[0].equals(SOURCE)) field = tokens[1];
```

Comment:
I personally find single line if statements not super readable.
Maybe keep using multiline if statements?